pkg/registry/experimental/controller/etcd/etcd.go

```diff
@@ -0,0 +1,123 @@
+/*
+Copyright 2014 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package etcd
+
+import (
+	"fmt"
+
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/api"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/api/rest"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/client"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/runtime"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/storage"
+
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/registry/controller"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/registry/controller/etcd"
+
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/expapi"
+)
+
+// PodStorage includes storage for pods and all sub resources
+type ContainerStorage struct {
+	ReplicationController *RcREST
+	Scaler                *ScalerREST
+}
+
+func NewStorage(s storage.Interface, k client.ConnectionInfoGetter) ContainerStorage {
+	rcRegistry := controller.NewRegistry(etcd.NewREST(s))
+
+	return ContainerStorage{
+		ReplicationController: &RcREST{},
+		Scaler:                &ScalerREST{registry: &rcRegistry},
+	}
+}
+
+type ScalerREST struct {
+	registry *controller.Registry
+}
+
+// LogREST implements GetterWithOptions
+var _ = rest.Patcher(&ScalerREST{})
+
+// New creates a new Scaler object
+func (r *ScalerREST) New() runtime.Object {
+	return &expapi.Scaler{}
+}
+
+func (r *ScalerREST) Get(ctx api.Context, name string) (runtime.Object, error) {
+	rc, err := (*r.registry).GetController(ctx, name)
+	if err != nil {
+	    return nil, err
+	}
+	return &expapi.Scaler{
+	    ObjectMeta: api.ObjectMeta {
+		Name: name,
+	    },
+	    Spec: expapi.ScalerSpec{ 
+		Replicas : &rc.Spec.Replicas,
+	    },
+	    Status: expapi.ScalerStatus{ 
+		Replicas : rc.Status.Replicas,
+		Selector : rc.Spec.Selector,
+	    },
+	}, nil
+}
+
+func (r *ScalerREST) Update(ctx api.Context, obj runtime.Object) (runtime.Object, bool, error) {
+	if obj == nil {
+		return nil,false, fmt.Errorf("Nil update passed to Scaler")
```

Comment:
Errors returned in a controller must always be API errors (notfound, bad request, invalid, etc).